modules/sqlite/integration/wp-includes/sqlite/class-wp-pdo-engine.php

```diff
@@ -0,0 +1,1500 @@
+<?php
+/**
+ * The main PDO extension class.
+ *
+ * @package performance-lab
+ * @since x.x.x
```

Comment:
See above, also generally for the documentation standards which aren't fully followed in this file and the other class files here either. To be fair, updating these docs would take a bit of time, so I'd like to understand a bit better:
* Where do these classes come from? Did you implement all of them for this module? Or are they from a previous (plugin) version of this module? Or are they ports of something external which you copied into here?
* For an eventual core merge, all these classes will need proper documentation anyway. So it's really just a matter of when to do this. I'd be open to not fixing everything in this PR so that we can launch something, but then I'd say it should be done in a follow-up PR.

WDYT?

Reply:
> Where do these classes come from? Did you implement all of them for this module? Or are they from a previous (plugin) version of this module? Or are they ports of something external which you copied into here?

The code for these files was copied from https://github.com/aaemnnosttv/wp-sqlite-db/blob/master/src/db.php, which is a fork of the original work on  the [sqlite-integration](https://wordpress.org/plugins/sqlite-integration/) plugin 👍
I split the single file to multiple files (one per class), fixed all coding-standards issues, converted it to work as a module, and I'm constantly trying to improve the quality of the code, bring it up to speed, do lots of performance tweaks, code-readability tweaks and so on.

> For an eventual core merge, all these classes will need proper documentation anyway. So it's really just a matter of when to do this. I'd be open to not fixing everything in this PR so that we can launch something, but then I'd say it should be done in a follow-up PR.

I agree 100%! I took the original code from [wp-swlite-db](https://github.com/aaemnnosttv/wp-sqlite-db), and I am slowly adding docs for all things missing. I already added `@param` docs to all methods and I'm slowly adding better documentation to all methods there. However, adding these docs is a huge task that takes a while. 
I think that these missing inline docs should not be considered a blocker here. We can merge the module in the plugin even with missing function/method docs, and we can keep working on them together so that, at some point, everything will be properly documented and ready for an eventual merge in Core 👍

Reply:
@aristath Thanks for clarifying, that all makes sense and not making those docs a blocker sounds good to me. Let's open an issue to at least capture here that we eventually need to polish the docs on these classes further.